src/uipath/_cli/_evals/_runtime.py

```diff
@@ -513,11 +519,97 @@ def _get_and_clear_execution_data(
 
         return spans, logs
 
+    async def _apply_model_settings_override(self) -> str | None:
+        """Apply model settings override if specified.
+
+        Returns:
+            Modified entrypoint path if settings were overridden, otherwise None
+        """
+        console = ConsoleLogger()
+        console.info(f"Checking model settings override with ID: '{self.context.model_settings_id}'")
+
+        # Skip if no model settings ID specified
+        if not self.context.model_settings_id or self.context.model_settings_id == "default":
+            return None
+
+        # Load evaluation set to get model settings
+        evaluation_set, _ = EvalHelpers.load_eval_set(self.context.eval_set or "")
+        if not hasattr(evaluation_set, 'model_settings') or not evaluation_set.model_settings:
+            console.warning("No model settings available in evaluation set")
+            return None
+
+        # Find the specified model settings
+        target_model_settings = next(
+            (ms for ms in evaluation_set.model_settings if ms.id == self.context.model_settings_id),
+            None
+        )
+
+        if not target_model_settings:
+            logger.warning(f"Model settings ID '{self.context.model_settings_id}' not found in evaluation set")
+            return None
+
+        console.info(f"Found model settings: model='{target_model_settings.model_name}', temperature='{target_model_settings.temperature}'")
```

Comment:
Please use logger, console is a CLI concept (imagine we extract the evals engine to another library)